src/ed.cpp

```diff
@@ -249,13 +199,17 @@ bool srvQuery(ed_msgs::Query::Request& req, ed_msgs::Query::Response& res)
     std::vector<ed::Idx> property_idxs;
     for(std::vector<std::string>::const_iterator it = req.properties.begin(); it != req.properties.end(); ++it)
     {
+        // ToDo: is this thread safe?
         const ed::PropertyKeyDBEntry* entry = ed_wm->getPropertyKeyDBEntry(*it);
         if (entry)
             property_idxs.push_back(entry->idx);
     }
 
-    const std::vector<unsigned long>& entity_revs = ed_wm->world_model()->entity_revisions();
-    const std::vector<ed::EntityConstPtr>&  entities = ed_wm->world_model()->entities();
+    // Make a copy of the WM, to keep it thead safe
+    ed::WorldModel wm = *ed_wm->world_model();
+
```

Comment:
I'd remove this line, as 209 is directly intended for line 211 and further

Reply:
Done